src/memex/storage.py

```diff
@@ -120,15 +120,17 @@ def create_annotation(request, data):
     annotation = models.Annotation(**data)
     annotation.created = created
     annotation.updated = updated
-    request.db.add(annotation)
 
-    models.update_document_metadata(
+    document = models.update_document_metadata(
         request.db,
         annotation.target_uri,
         document_meta_dicts,
         document_uri_dicts,
         created=created,
         updated=updated)
+    annotation.document_id = document.id
+
+    request.db.add(annotation)
```

Comment:
There's no need to move the call to `.add`, and it's probably easier to see that it's been called if it stays closest to the call to the `Annotation` constructor.